target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl838x.c

```diff
@@ -1805,20 +1806,14 @@ irqreturn_t rtl838x_switch_irq(int irq, void *dev_id)
 
 int rtl838x_smi_wait_op(int timeout)
 {
-	unsigned long end = jiffies + usecs_to_jiffies(timeout);
+	int ret = 0;
+	u32 val;
 
-	while (1) {
-		if (!(sw_r32(RTL838X_SMI_ACCESS_PHY_CTRL_1) & 0x1))
-			return 0;
+	ret = readx_poll_timeout(sw_r32, RTL838X_SMI_ACCESS_PHY_CTRL_1, val, val & 0x1, 20, timeout);
+	if (ret)
+		pr_err("%s: timeout\n", __func__);
```

Comment:
is it possible to use dev_err as we are changing this?

Reply:
I'm generally much in favor, however, we use dev_err almost nowhere, also, shouldn't be netdev_err in that case? anyway, I didn't want to touch too much in this commit and just keep everything 'as is'. Except a 'treewide' *_err etc in not too long a time anyway. Personally, I'd even remove this entire pr_err anyway, and just return readx_poll_timeout.

Having said that, even that'll change, once we drop sw_r32.

So yes, we should, but I"m a bit hesitant, as otherwise, we spend up too much time refactoring on small cleanups :)

Reply:
Enough reason to keep the change as is.